lib/improver/spotdata/spot_extraction.py

```diff
@@ -202,14 +204,17 @@ def process(self, neighbour_cube, diagnostic_cube, new_title=None):
         spotdata_cube = data_cubes.merge_cube()
 
         # Copy attributes from the diagnostic cube that describe the data's
-        # provenance, and update title if specified
+        # provenance
         spotdata_cube.attributes = diagnostic_cube.attributes
         spotdata_cube.attributes['model_grid_hash'] = (
             neighbour_cube.attributes['model_grid_hash'])
-        if new_title is not None:
-            spotdata_cube.attributes["title"] = new_title
-        else:
-            spotdata_cube.attributes.pop("title", None)
+
+        # Remove grid attributes and update title
```

Comment:
Should we update a docstring to say the MOSG_GRID_ATTRIBUTES will be removed?

Reply:
Also is there a test which check the MOSG_GRID_ATTRIBUTES are removed?

Reply:
Yes - but implicitly.

If you look at the changes I've made to the spot extraction unit tests, you'll see I've added an `expected_attributes` member to the class.  These attributes are a copy of what's on the input diagnostic cube, with the grid attributes removed.  This tests removal because the `diagnostic_xy` cube has grid attributes on it, but wouldn't be an adequate test if those weren't present.

I'll add a comment to the setup of those test cubes saying why these attributes are essential.